catalog/src/test/java/org/killbill/billing/catalog/io/TestVersionedCatalogLoader.java

```diff
@@ -126,4 +128,43 @@ public void testLoad() throws IOException, SAXException, InvalidConfigException,
         dt = new DateTime("2011-03-03T00:00:00+00:00");
         Assert.assertEquals(it.next().getEffectiveDate(), dt.toDate());
     }
+
+    @Test(groups = "fast")
+    public void testLoadCatalogFromClasspathResourceFolder() throws CatalogApiException {
+        final VersionedCatalog c = loader.loadDefaultCatalog("SpyCarBasic.xml");
+        Assert.assertEquals(c.size(), 1);
+        DateTime dt = new DateTime("2013-02-08T00:00:00+00:00");
+        Assert.assertEquals(c.getEffectiveDate(), dt.toDate());
+        Assert.assertEquals(c.getCatalogName(), "SpyCarBasic");
+    }
+
+    @Test(groups = "fast", expectedExceptions = CatalogApiException.class)
+    public void testLoadCatalogFromClasspathResourceBadFolder() throws CatalogApiException {
+        loader.loadDefaultCatalog("SpyCarCustom.xml");
+    }
+
+    @Test(groups = "fast")
+    public void testLoadCatalogFromInsideResourceFolder() throws CatalogApiException, URISyntaxException, IOException {
+        final VersionedCatalog c = loader.loadDefaultCatalog("com/acme/SpyCarCustom.xml");
+        Assert.assertEquals(c.size(), 1);
+        DateTime dt = new DateTime("2015-10-04T00:00:00+00:00");
+        Assert.assertEquals(c.getEffectiveDate(), dt.toDate());
+        Assert.assertEquals(c.getCatalogName(), "SpyCarCustom");
+    }
+
+    @Test(groups = "fast", expectedExceptions = CatalogApiException.class)
+    public void testLoadCatalogFromInsideResourceWithBadFolderName() throws CatalogApiException {
+        loader.loadDefaultCatalog("com/acme2/SpyCarCustom.xml");
+    }
+
+    @Test(groups = "fast")
+    public void testLoadCatalogFromExternalFile() throws CatalogApiException, IOException, URISyntaxException {
+        URL originURL = Resources.getResource("SpyCarBasic.xml");
+        File originFile = new File(originURL.toURI());
+        File destinationFile = new File("/var/tmp/SpyCarBasic.xml");
+        destinationFile.deleteOnExit();
+        Files.copy(originFile, destinationFile);
+        final VersionedCatalog c = loader.loadDefaultCatalog(destinationFile.toURI().toString());
```

Comment:
This test probably also passes even before the code change because the name of the xml file is the same (bug reported on the mailing-list). Can you change the name (e.g. `SpyCarBasicRelocated.xml`), just to ensure the right catalog is loaded?